build_tools/cmake/ctest_all.sh

```diff
@@ -15,8 +15,16 @@ set -euo pipefail
 
 BUILD_DIR="$1"
 
+get_default_parallel_level() {
+  if [[ "$(uname)" == "Darwin" ]]; then
+    echo $(sysctl -n hw.physicalcpu)
+  else
+    echo $(nproc)
+  fi
+}
```

Comment:
Wouldn't the equivalent be `hw.logicalcpu`?
```suggestion
get_default_parallel_level() {
  if [[ "$(uname)" == "Darwin" ]]; then
    echo $(sysctl -n hw.logicalcpu)
  else
    echo $(nproc)
  fi
}
```

It would also be nice if this could be a one line command shared between platforms :/. I'd like to keep scripts like this one as simple as possible.

Reply:
Actually, you are right. `nproc` returns logical core count. Though on Apple silicon, these two are the same.

Reply:
I've updated it to make it a oneliner. :)

Reply:
Uh, yeah I liked the multiple lines better... If we want this used in multiple scripts we could move it to some helper function

Reply:
So, should I revert back to the original form? I don't have a strong opinion here.

Reply:
I would prefer that (with the nit to remember to quote your command expansions)

Reply:
Reverted to the old form.